typescript/ws-protocol/src/FoxgloveServer.ts

```diff
@@ -3,25 +3,33 @@ import { EventEmitter, EventNames, EventListener } from "eventemitter3";
 
 import { ChannelId, StatusLevel } from ".";
 import {
+  BinaryOpcode,
   Channel,
+  ClientChannel,
   ClientMessage,
-  SubscriptionId,
-  ServerMessage,
-  BinaryOpcode,
+  ClientPublish,
   IWebSocket,
+  ServerCapability,
+  ServerMessage,
+  SubscriptionId,
 } from "./types";
 
 type EventTypes = {
+  error: (error: Error) => void;
+
   subscribe: (channel: ChannelId) => void;
   unsubscribe: (channel: ChannelId) => void;
-  error: (error: Error) => void;
+  message: (event: ClientPublish) => void;
+  advertise: (channel: ClientChannel) => void;
+  unadvertise: (channel: ChannelId) => void;
```

Comment:
I wonder if naming these `clientAdvertise`/`clientUnadvertise` would help reduce confusion? On the other hand, we didn't name the other ones `clientSubscribe`/`clientUnsubscribe`....